DeepSpeech.ipynb

```diff
@@ -1458,49 +1459,51 @@
-    "                                  'dropout_rate': dropout_rate_placeholder }),\n",
-    "                            'outputs': exporter.generic_signature(\n",
-    "                                { 'outputs': decoded})})\n",
-    "    model_exporter.export(export_dir, tf.constant(export_version), session)\n",
-    "\n",
-    "print 'Model exported at %s' % (export_dir)"
+    "if export_dir != None:\n",
+    "    with tf.device('/cpu:0'):\n",
+    "        model_exporter = exporter.Exporter(saver)\n",
+    "\n",
+    "        # Run inference\n",
+    "        # Input tensor will be of shape [batch_size, n_steps, n_input + 2*n_input*n_context]\n",
+    "        input_tensor = tf.placeholder(tf.float32, [None, None, n_input + 2*n_input*n_context])\n",
+    "\n",
+    "        # Calculate input sequence length. This is done by tiling n_steps, batch_size times.\n",
+    "        # If there are multiple sequences, it is assumed they are padded with zeros to be of\n",
+    "        # the same length.\n",
+    "        n_items  = tf.slice(tf.shape(input_tensor), [0], [1])\n",
+    "        n_steps = tf.slice(tf.shape(input_tensor), [1], [1])\n",
+    "        seq_length = tf.tile(n_steps, n_items)\n",
+    "\n",
+    "        # Calculate the logits of the batch using BiRNN\n",
+    "        logits = BiRNN(input_tensor, tf.to_int64(seq_length))\n",
+    "\n",
+    "        # Beam search decode the batch\n",
+    "        logits = tf.transpose(logits, [1, 0, 2])\n",
+    "        decoded, _ = ctc_ops.ctc_beam_search_decoder(logits, seq_length)\n",
+    "        decoded = tf.convert_to_tensor(\n",
+    "            [tf.sparse_tensor_to_dense(sparse_tensor) for sparse_tensor in decoded])\n",
+    "\n",
+    "        # TODO: Transform the decoded output to a string\n",
+    "\n",
+    "        # Restore variables from training checkpoint\n",
+    "        # TODO: This restores the most recent checkpoint, but if we use validation to counterract\n",
+    "        #       over-fitting, we may want to restore an earlier checkpoint.\n",
+    "        checkpoint = tf.train.get_checkpoint_state(checkpoint_dir)\n",
+    "        saver.restore(session, checkpoint.model_checkpoint_path)\n",
+    "        print 'Restored checkpoint at training epoch %d' % (int(checkpoint.model_checkpoint_path.split('-')[1]) + 1)\n",
+    "\n",
+    "        # Initialise the model exporter and export the model\n",
+    "        model_exporter.init(session.graph.as_graph_def(),\n",
+    "                            named_graph_signatures = {\n",
+    "                                'inputs': exporter.generic_signature(\n",
+    "                                    { 'input': input_tensor,\n",
+    "                                      'dropout_rate': dropout_rate_placeholder }),\n",
+    "                                'outputs': exporter.generic_signature(\n",
+    "                                    { 'outputs': decoded})})\n",
+    "        shutil.rmtree(export_dir)\n",
```

Comment:
If we change the behavior to be to not export by default, I don't think we need the buried rm call :)